### PR TITLE
[EMCAL-501] modified timing in Digitizer

### PR DESCRIPTION
**Pull request by aknospe**
The modifications are
1.) accounting for the 4 different phases when dividing the energy into time bins
2.) introducing the 700 ns delay (the value can be modified)
3.) simulating noise digits at the beginning of each 1500 ns period
4.) accumulating digits during the 1500 ns period
5.) summing digits and passing them to the DigitizerSpec at the end of each 1500 ns period

### CODE REVIEW

**Comment by sawenzel** [comment]
What is the status here? Is this ready or still being developed?

**Comment by mfasDa** [comment]
@sawenzel The PR is now ready to be merged!

File: DataFormats/Detectors/EMCAL/src/DataFormatsEMCALLinkDef.h [hunk 1]
```diff
@@ -19,12 +19,16 @@
...
 #pragma link C++ class o2::emcal::Digit + ;
 #pragma link C++ class o2::emcal::Cluster + ;
+#pragma link C++ class o2::emcal::MCLabel + ;
 
 #pragma link C++ class std::vector < o2::emcal::TriggerRecord > +;
 #pragma link C++ class std::vector < o2::emcal::Cell > +;
 #pragma link C++ class std::vector < o2::emcal::Digit > +;
 #pragma link C++ class std::vector < o2::emcal::Cluster > +;
 
+#include "SimulationDataFormat/MCTruthContainer.h"
```
Comment (shahor02) [inline DataFormats/Detectors/EMCAL/src/DataFormatsEMCALLinkDef.h]: I think it is better to have this include in your MCLabel class

File: Detectors/EMCAL/simulation/include/EMCALSimulation/SimParam.h [hunk 2]
```diff
@@ -82,32 +76,51 @@ class SimParam
   void setECPrimaryThreshold(Float_t val) { mECPrimThreshold = val; }
   Float_t getECPrimaryThreshold() const { return mECPrimThreshold; }
 
+  void setSignalDelay(Float_t val) { mSignalDelay = val; }
```
Comment (shahor02) [inline Detectors/EMCAL/simulation/include/EMCALSimulation/SimParam.h]: The setters are not compatible with the way ConfigurableParamHelper works:
the ``SimParam::Instance()`` returns ``const ref``, so the setters are useless (unless you clone the SimParam). The proper way to set the singletone data members is via static method, e.g.
``o2::conf::ConfigurableParam::updateFromString("EMCSimParam.mA=1")`` or from the ini file with similar entries via ``o2::conf::ConfigurableParam::updateFromFile(inifile)``

@sawenzel , please correct me if I am wrong.

File: DataFormats/Detectors/EMCAL/src/DataFormatsEMCALLinkDef.h (hunk 1 above)
Comment (sawenzel) [reply DataFormats/Detectors/EMCAL/src/DataFormatsEMCALLinkDef.h]: I think it's ok to have it here. We have done this elsewhere as well and MCLabel probably does not depend on the Container.

File: Detectors/EMCAL/simulation/include/EMCALSimulation/SimParam.h (hunk 2 above)
Comment (sawenzel) [reply Detectors/EMCAL/simulation/include/EMCALSimulation/SimParam.h]: That's correct. We can discuss other ways if there is a use-case not currently covered.

**Comment by sawenzel** [comment]
Any news here? I think this only needs minor polishing.

**Comment by mfasDa** [comment]
@sawenzel I think this PR could be merged now

**Comment by aknospe** [comment]
Yes, I think so too.

**Pull request merged**